### PR TITLE
Add minimal PWA functionality

### LINKED ISSUE #52: Add Minimal Progressive Web App (PWA) functionality
## Description

We'll need to turn our app into a PWA to give users the option to "install" it as an app. The prerequisite for this is creating a [manifest](https://github.com/vercel/next.js/blob/canary/examples/progressive-web-app/public/manifest.json) file and linking it in our [head](https://github.com/vercel/next.js/blob/canary/examples/progressive-web-app/pages/_app.tsx).

By default, this prompt will appear on the first page the user sees on our site. I think it makes more sense to show the PWA install prompt on the dashboard page; the first page the user sees that they'll come back to every day. To do this, we'll need to catch an event and change the time we present the prompt. A quick guide can be found here: https://web.dev/learn/pwa/installation-prompt/#the-beforeinstallprompt-event

## Todos

These points are a rough guideline. **Please** feel free to discuss with others on the team about the best way to design this component!

- [ ] Add a `manifest.json` file to the public folder and customize it to suit our repo
- [ ] Link to the manifest in the `<head`
- [ ] Handle the before install prompt and only re-prompt it on the dashboard page

## Time estimate

tbd

### PR DESCRIPTION
**Pull request by code-arman**
## Status:
:rocket: Ready

## Description
Adds PWA functionality so that it can be downloaded from chrome. You will see this button when testing it out 
Screen Shot 2023-03-03 at 3 25 17 AM

Fixes #52
## Todos
- Fix logos (i took a random logo and resized it)
- Figure out correct styling/positioning for button
- Figure out what offline page is supposed to look like

## Screenshots

The button only shows on non-mobile views
Screen Shot 2023-03-09 at 3 43 09 AM

PWA version doesn't show button:
Screen Shot 2023-03-09 at 3 43 15 AM

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,3 +35,11 @@ yarn-error.log*
...
 *.tsbuildinfo
 next-env.d.ts
+
+# PWA files
+**/public/sw.js
+**/public/workbox-*.js
+**/public/worker-*.js
+**/public/sw.js.map
+**/public/workbox-*.js.map
+**/public/worker-*.js.map
```
Comment (AndrewLester) [inline .gitignore]: new line at end of file

**Review by AndrewLester — APPROVED** [review]
This all looks good pending one small thing, and we'll fix conflicts for you.

**Review by AndrewLester — APPROVED** [review]
Thanks

**Review by ericcccsliu — APPROVED** [review]
LGTM

**Pull request merged**